py3status/modules/taskwarrior.py

```diff
@@ -55,16 +58,27 @@
     For more information, see https://taskwarrior.org/docs/udas.html
 
 format_annotation placeholders:
+    {index} index number, eg 1
     {entry} entry date, eg 20171021T010203Z
     {description} description, eg Tell friends about py3status
 
 format_tag placeholders:
-    {tag} tag name, eg home, work
+    {index} index number, eg 1
+    {name} tag name, eg home, work
 
 Color thresholds:
-    count: print color based on number of tasks
-    id: print color based on id number
-    urgency: print color based on urgency measure
+    format:
+        task: color threshold of number of tasks
+    format_task:
+        task_index: color threshold of task index
+        id: color threshold of task id
+        urgency: color threshold of urgency measure
+        tag: color threshold of number of tags
+        annotation: color threshold of number of annotations
+    format_tag:
+        tag_index: color threshold of tag index
+    format_annotation:
+        annotation_index: color threshold of annotation index
```

Comment:
I'm just running this by you. It's not really an issue, but could be a bit nuisance...
Here, I'm playing with lot of thresholds.

```
thresholds = {
    'annotation': [(1, 'good'), (2, 'degraded'), (3, 'bad')],
    'tag': [(1, 'bad'), (2, 'degraded'), (3, 'good')],
    'task': [(1, 'bad'), (2, 'degraded'), (3, 'good')],

    'annotation_index': [(1, 'good'), (2, 'degraded'), (3, 'bad')],
    'tag_index': [(1, 'bad'), (2, 'degraded'), (3, 'good')],
    'task_index': [(1, '#ffaa00'), (2, '#aaff00'), (3, '#aaccff')],
}
```

I'm showing all, but let's just focus on tags. If users want to colorize tags using thresholds so it just can have different tag colors based on index. They will use this.
```
format_annotation = '\?color=annotation_index {index} {description}'
format_tag = '\?color=tag_index {index} {name}'
format_task = '\?color=task_index {index} {name}'
```

Ideal...
```
format_annotation = '\?color=index {index} {description}'
format_tag = '\?color=index {index} {name}'
format_task = '\?color=index {index} {name}'
```

I can keep it a bit easier... Making them all use same `index` threshold, but then they will have to accept one threshold for all. I'm just playing around... Seeing if we can push py3status beyond its limit... Lol.

This is for inner `format_*` inside outer `format_*` where thresholds are not likely needed to be exposed to the outer `format_*`, but to keep names consistent through all `format_*`.

You can see I was able to use same `{index}` for placeholders inside the formats, but not the thresholds. I just want to hear what you have to say about this... And see if we can make `thresholds` limited to `formats` inside the code...
because once it goes inner `format_*`, the thresholds names become silly due to same format_* names.

This is fucking hard for users... to be explicit. Ah forget it. :-\

```
thresholds = {
    'annotation': [(1, 'good'), (2, 'degraded'), (3, 'bad')],
    'tag': [(1, 'bad'), (2, 'degraded'), (3, 'good')],
    'task': [(1, 'bad'), (2, 'degraded'), (3, 'good')],

    'format_annotation': {
        'index': {[(1, 'good'), (2, 'degraded'), (3, 'bad')]},
    },
    'format_tag': {
        'index': {[(1, 'bad'), (2, 'degraded'), (3, 'good')]},
    }
    'format_task': {
        'index': {[(1, '#ffaa00'), (2, '#aaff00'), (3, '#aaccff')]},
    }
}
```